routes/mux/upload.ts

```diff
@@ -0,0 +1,82 @@
+import Mux from '@mux/mux-node'
+import { useCORS } from 'nitro-cors'
+
+const { Video } = new Mux()
+
+type HTTPMethod =
+  | 'GET'
+  | 'HEAD'
+  | 'PATCH'
+  | 'POST'
+  | 'PUT'
+  | 'DELETE'
+  | 'CONNECT'
+  | 'OPTIONS'
+  | 'TRACE'
+
+export default defineEventHandler(async (event) => {
+  const corsOptions = {
```

Comment:
What was the issue you were getting when defining these properties as route rules?

Reply:
[This issue](https://github.com/unjs/nitro/issues/1095) seems to encapsulate what you're saying. Do you feel like it makes sense to go ahead with one of the approaches in the comments that don't require another package, or you feel confident in the current impl?

Reply:
https://github.com/michealroberts/nitro-cors

the package is well maintained, would prefer to leave this implementation